File: Producto/Backend/users-service/src/main/java/cl/duoc/pichangapp/users_service/dto/PerfilPublicoDTO.java
package cl.duoc.pichangapp.users_service.dto;

/**
 * Perfil público de un usuario: lo que se muestra a OTROS usuarios.
 * No expone id ni contraseña. El correo se incluye como clave única de identidad.
 */
public record PerfilPublicoDTO(
        String correo,
        String nombre,
        String apellido,
        Integer karmaScore,
        String categoriaKarma,
        Boolean historialVisible
) {}

File: Producto/front-end/PichangApp/app/src/main/java/cl/duoc/pichangapp/data/model/PerfilPublicoDto.kt
package cl.duoc.pichangapp.data.model

import androidx.compose.runtime.Stable

/**
 * Perfil público de otro usuario (lo que devuelve el backend en
 * GET /api/v1/users/buscar y /api/v1/users/perfil-publico/{correo}).
 * No incluye correo ni id por privacidad.
 */
@Stable
data class PerfilPublicoDto(
    val correo: String? = null,
    val nombre: String? = null,
    val apellido: String? = null,
    val karmaScore: Int = 0,
    val categoriaKarma: String? = null,
    val historialVisible: Boolean = true
)
